modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/rest-openapi.yaml

```diff
@@ -1759,5 +1792,624 @@ paths:
+                  required: true
+                  schema:
+                      description:
+                          WishList Id
+                      format: int64
+                      type: integer
+                  style: simple
+                - in: query
+                  name: accountId
+                  required: false
+                  schema:
+                      description:
+                          Account Id
+                      example: 30130
+                      format: int64
+                      minimum: 0
+                      type: integer
+            requestBody:
+                content:
+                    application/json:
+                        schema:
+                            $ref: "#/components/schemas/WishListItem"
+                    application/xml:
+                        schema:
+                            $ref: "#/components/schemas/WishListItem"
+                required: true
+            responses:
+                201:
+                    content:
+                        application/json:
+                            schema:
+                                $ref: "#/components/schemas/WishListItem"
+                        application/xml:
+                            schema:
+                                $ref: "#/components/schemas/WishListItem"
+                    description:
+                        Created
+                400:
+                    description:
+                        Invalid input
+                401:
+                    content:
+                        application/json:
+                            schema:
+                                $ref: "#/components/schemas/Error"
+                        application/xml:
+                            schema:
+                                $ref: "#/components/schemas/Error"
+                    description:
+                        Authentication information is missing or invalid
+                404:
+                    content:
+                        application/json:
+                            schema:
+                                $ref: "#/components/schemas/Error"
+                        application/xml:
+                            schema:
+                                $ref: "#/components/schemas/Error"
+                    description:
+                        The specified resource was not found
+                500:
+                    content:
+                        application/json:
+                            schema:
+                                $ref: "#/components/schemas/Error"
+                        application/xml:
+                            schema:
+                                $ref: "#/components/schemas/Error"
+                    description:
+                        Unexpected error
+            summary: Creates a WishList Item.
+            tags: ["WishListItem"]
+    /channels/{channelId}/wishlists/{wishListId}/wishlist-items/{wishListItemId}:
```

Comment:
after reviewing this with @luca-pellizzon I believe you don't need to pass neither the channel id nor the wishlistId, you can find those info from the wishlistitem, you can fetch the wishlist from the item directly and you can fetch the channel with channel service looking for sitegroup id

Reply:
@andrea-ale-sbarra I removed both channelId and wishListId but I think there is a bug that stopping me from proceeding. 
The getCommerceChannelByGroupId method ends up calling commerceChannelLocalService.getCommerceChannel(
group.getClassPK()) which expects channelId but instead we are passing groupId

My code is as follows:
<<<<<<<<<<<<
public WishListItem getWishListItem(
			Long accountId, Long wishListItemId)
throws Exception {

CommerceWishListItem commerceWishListItem =
	_commerceWishListItemService.getCommerceWishListItem(
		wishListItemId);

CommerceChannel commerceChannel =
	_commerceChannelLocalService.getCommerceChannelByGroupId(
		commerceWishListItem.getGroupId());

CommerceContext commerceContext = _commerceContextFactory.create(
	commerceChannel.getCompanyId(), commerceChannel.getGroupId(),
	commerceChannel.getUserId(), 0,
	_getCommerceAccountId(accountId, commerceChannel));

return _toWishListItem(commerceWishListItem, commerceContext);
}
>>>>>>>>>>>>

Reply:
@andrea-ale-sbarra 
Please note that the code above is not in this PR